### PR TITLE
5704 memory leak in web3.eth.contract

### LINKED ISSUE #5704: Memory leak in web3.eth.Contract
### This is not a Bug Report, Feature Request, or related to Documentation

- [X] I have searched the existing issues

### Is there an existing issue for this?

- [X] I have searched the existing issues

### What's up?

I have made a fully synced ethereum node using binaries (heimdall and bor) and use its RPC point to get blockchain's data. When i start processing blocks, after some time the heap is out of memory. Upon investigating I got to know that this issue is because of `web3.eth.Contract`.

### Environment
Operating system: Mac
bor version: v0.2.16
heimdall version: v0.2.11
web3-eth-contract : v1.8.1

### Steps to reproduce
You can run the following sample code with
`````
const Contract = require("web3-eth-contract");
export const getContract = (tokenAddress: string) => {
  Contract.setProvider(web3.currentProvider);
  const contract = new Contract([],'0xc14ca0805194dc4a1319dcad80352d8033763b92');
  return contract;
};
`````

### Examples/References

I have tried different approaches specially from these two links https://github.com/web3/web3.js/issues/3042 https://github.com/web3/web3.js/pull/3866 but my issue persists

### PR DESCRIPTION
**Pull request by avkos**
## Description
Fixes #5704 

So the problem was a new `SubscriptionManager` which was created every time when making a new instance of the Contract class because, by default, web3 didn't create the `registeredSubscriptions` object. So I've added the object by default. And now it always reuses the same `SubscriptionManager` in the contract class.

Also, I've removed the array of contracts in the web3 package. Because we don't need it. When we set the provider in one of the contracts or in web3, it affects all instances.

After these changes memory leak disappears.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist for 4.x:

- [ ] I have selected the correct 4.x base branch.
- [ ] Within the description, the feature or issue is discussed in detail or an issue is linked.   
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have added any required tests for the changes I made 
- [ ] I have made corresponding changes to the documentation
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran `yarn` successfully
- [ ] I ran `yarn lint` successfully
- [ ] I ran `yarn build:web` successfully
- [ ] I ran `yarn test:unit` successfully
- [ ] I ran `yarn test:integration` successfully
- [ ] I ran `compile:contracts` successfully
- [ ] I have tested my code.
- [ ] I have updated the corresponding `CHANGELOG.md` file in the packages I have edited.

### CODE REVIEW

File: packages/web3/src/web3.ts [hunk 1]
```diff
@@ -91,16 +88,6 @@ export class Web3 extends Web3Context<EthExecutionAPI> {
 				} else {
 					super(jsonInterface, self.getContextObject());
 				}
-
```
Comment (jdevcs) [inline packages/web3/src/web3.ts:94]: Also update change log for this.

**Review by jdevcs — APPROVED** [review]
lgtm, except a minor change log work.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1138,3 +1138,13 @@ should use 4.0.1-alpha.0 for testing.
 ## [Unreleased]
 
 -   Added rpc exception codes following eip-1474 as an experimental feature (if `useRpcCallSpecification` at `enableExperimentalFeatures` is `true`) (#5525)
+
+#### web3
+
+##### Removed
+
+-   Private static `_contracts:Contract[]` and static `setProvider` function was removed
```
Comment (luu-alex) [inline CHANGELOG.md]: Add PR/Issue number

**Review by luu-alex — APPROVED** [review]

**Review by nikoulai — APPROVED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (avkos) [reply CHANGELOG.md]: done

File: packages/web3/src/web3.ts (hunk 1 above)
Comment (avkos) [reply packages/web3/src/web3.ts:94]: done

**Pull request merged**